packages/core/src/internal/newTask.js

```diff
@@ -0,0 +1,134 @@
+import deferred from '@redux-saga/deferred'
+import * as is from '@redux-saga/is'
+import { TASK, TASK_CANCEL } from '@redux-saga/symbols'
+import { assignWithSymbols, check, createSetContextWarning } from './utils'
+import { addSagaStack, sagaStackToString } from './error-utils'
+import forkQueue from './forkQueue'
+
+export default function newTask(env, mainTask, parentContext, parentEffectId, meta, isRoot, cont) {
```

Comment:
The interface of `newTask()` is very much like `proc()`.

`proc(env, iterator, parentContext, parentEffectId, meta, isRoot, cont)`

Reply:
Just posting both here for convenience:
```js
function newTask(env, mainTask, parentContext, parentEffectId, meta, isRoot, cont)
function proc(env, iterator, parentContext, parentEffectId, meta, isRoot, cont)
```

They indeed seem to relate to each other really closely, wondering if we can do smth about it.